addons/survey/static/src/js/survey_form.js

```diff
@@ -617,23 +617,25 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
         var minDateData = $dateGroup.data('mindate')
         var maxDateData = $dateGroup.data('maxdate')
 
-        var datetimepickerFormat = time.getLangDateFormat();
-        if ($dateGroup.find('input').data('questionType') === 'datetime') {
-            datetimepickerFormat = time.getLangDatetimeFormat();
-        } else {
+        var datetimepickerFormat = $dateGroup.find('input').data('questionType') === 'datetime'
+                                        ? datetimepickerFormat = time.getLangDatetimeFormat()
```

Comment:
why `datetimepickerFormat =` again ?

As you now use multiple time `$dateGroup.find('input').data('questionType')`, make it a variable.
so you can put var datetimepickerFormat in one line (because short conditional assignment is cool enough)

Reply:
Indeed, I made the changes!